craft_store/store_client.py

```diff
@@ -98,6 +99,9 @@ def __init__(
         self._endpoints = endpoints
 
         self._auth = Auth(application_name, base_url, environment_auth=environment_auth)
+        self._ephemeral_login = (
+            environment_auth is not None and os.getenv(environment_auth) is not None
```

Comment:
If `environment_auth` is set to an empty string or something else that's not a valid environment variable name we get `None` so we're good here.